cypress/support/utils.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Utility function to escape arguments for safe inclusion in shell commands.
+ */
+export function escapeForShellCommand(args: string) {
+  const argsArray = args.split(' ');
+  let argsString = '';
+
+  for (let i = 0; i < argsArray.length; i++) {
+    argsString += ' ';
+    if (argsArray[i].startsWith('-') && i < argsArray.length - 1) {
+      argsString += argsArray[i] + ' ' + `'${argsArray[i + 1]}'`;
```

Comment:
@himdel The util function is pretty barebones right now anmd while it is meeting the needs for creating/deleting collections and namespaces I anticipate we will need to expand its capabilities as we add more complex commands. Do you think using strings with apostrophe is something that we will come across as we write more tests? 
Also, I'm thinking that's probably something that can be addressed if I replaced single quotes around arguments with double quotes. What do you think?

Reply:
Just for background, the new dependency `shell-escape-tag` (its underlying dependency any-shell-escape) requires polyfills to work. I discussed it with @jamestalton and we agreed it may be worth using a function of our own. If there are several more scenarios that are going to add too much complexity to this util function, then it's definitely worth looking into using a separate library.

Reply:
Ah, well, if the goal is to remove it.. My concern was that escaping shell properly is complicated, so not having to solve it sounds ..safer. But then again, the stakes are low here, arbitrary shell execution on github actions runner is probably the worst case screnario, and it can only be done by people who contribute code and thus have that ability anyway.

We do need to support spaces (repository names can contain spaces), apostrophes are too easy to introduce with english words, so probably those too.. maybe exclamation marks. The rest should be mostly fine, as long as nobody starts testing if weird names work :).

So.. I guess something like ``const escape = (s) => `'${s.replaceAll(/['\\]/g, '\\$&')}'`;`` on all params except the command and the first param *could* work...

Reply:
@himdel Thanks for the regex recommendation! I altered it slightly since I'm surrounding the argument in double quotes. Instead of the apostrophe I escaped the exclamation mark. I needed to make an update (ES2021) in the tsconfig to be able to use `replaceAll`. 

I tested the updated code with the following command:
```
    cy.galaxykit('-i repository create', "this is Vidya's repository!");
```
The repo was created correctly:

![image](https://github.com/ansible/ansible-ui/assets/43621546/c250c5e7-8ae5-453b-8f75-d68d86dc199e)

Since `galaxykit` is being used for generating resources for tests, in most cases the names would have to be ones that are correctly accepted by the API. As for testing different kinds of names, that'll probably be a better candidate for component tests where we can verify validation. 

Agreed that escaping shell properly and accounting for all cases is complicated so if we find that we need to handle complex scenarios, it makes sense to discard this approach and look at alternative libraries.

Reply:
Sounds good :) 

One note ... if the UI is displaying `repository\!` (instead of `repository!`), there's more escaping going on than it should..

Reply:
Oh... except if you use double quotes now, many more characters are special inside double quotes .. so we'd also want to escape `$` and `` ` ``

Reply:
I see the same repository created in the UI if I run `galaxykit -s 'http://localhost:5001/api/automation-hub/' -u <username> -p <pw> repository create "this Vidya's repository\!"` from the terminal/

Reply:
@himdel okay, final iteration (hopefully 🙏 ) otherwise I'm going to use a library! I changed the regex again this time to `const newString = s.replace(/[\\]/g, '\\$&');`. And I'm able to create the repository:
```
cy.galaxykit('-i repository create', "thi$ i$ !Vidya's repository!");
```
![image](https://github.com/ansible/ansible-ui/assets/43621546/07d2b406-7b72-46ca-a4fa-1490744d7af0)

Reply:
Oh, looks like the simple solution works quite well :+1: 

... it will break for strings containing a double quote, but that's probably fine now :)